### DISCUSSION TITLE
How to input unpaired data after error correction?

### DISCUSSION THREAD

**Discussion by 1996xjm** [ablab/spades#865] [Q&A] [answered]
### Description of bug

I have a paired-end data of bacteria. After error correction using trimmomatic, I got four files: QD34_12_paired_1.fq.gz, QD34_12_paired_2.fq.gz, QD34_12_unpaired_1.fq.gz, QD34_12_unpaired_2.fq.gz. For unpaired fastq files, I don't know how to edit my YAML file. Is it right as the follow code show? Can spades recognize whether the unpaired file is forward or reverse?
>  ## YAML File
```
[{
	"orientation": "fr",
	"type": "paired-end",
	"left reads": ["/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_paired_1.fq.gz"],
	"right reads": ["/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_paired_2.fq.gz"]
},
{
	"type": "single",
	"single reads": ["/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_unpaired_1.fq.gz", "/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_unpaired_2.fq.gz"]
}]
```

### spades.log

no

### params.txt

Command line: /home/xjm/miniconda3/envs/spades/SPAdes-3.15.3-Linux/bin/spades.py	--isolate	--dataset	/userData/xjm/genome_seq_Po/patch2/spades_output/QD34_12/QD34_12.yaml	-t	28	-m	128	-o	/userData/xjm/genome_seq_Po/patch2/spades_output/QD34_12	

System information:
  SPAdes version: 3.15.3
  Python version: 3.7.8
  OS: Linux-4.15.0-161-generic-x86_64-with-debian-buster-sid

Output dir: /userData/xjm/genome_seq_Po/patch2/spades_output/QD34_12
Mode: ONLY assembling (without read error correction)
Debug mode is turned OFF

Dataset parameters:
  Isolate mode
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_paired_1.fq.gz']
      right reads: ['/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_paired_2.fq.gz']
      interlaced reads: not specified
      single reads: not specified
      merged reads: not specified
    Library number: 2, library type: single
      left reads: not specified
      right reads: not specified
      interlaced reads: not specified
      single reads: ['/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_unpaired_1.fq.gz', '/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_unpaired_2.fq.gz']
      merged reads: not specified
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned OFF
  MismatchCorrector will be SKIPPED
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /userData/xjm/genome_seq_Po/patch2/spades_output/QD34_12/tmp
  Threads: 28
  Memory limit (in Gb): 128

### SPAdes version

3.15.3

### Operating System

Linux-4.15.0-161-generic-x86_64-with-debian-buster-sid

### Python Version

3.7.8

### Method of SPAdes installation

manual

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment, marked as answer]
You can simply add your single reads as "single reads" portion of your paired-end library. Orientation of single reads does not make any sense, as it's single read

**Reply by 1996xjm** [reply]
Can I write YAML file like  this?
```
[{
	"orientation": "fr",
	"type": "paired-end",
	"left reads": ["/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_paired_1.fq.gz"],
	"right reads": ["/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_paired_2.fq.gz"],
"single reads": ["/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_unpaired_1.fq.gz", "/userData/xjm/genome_seq_Po/patch2/trimmomatic_output-1/QD34_12/QD34_12_unpaired_2.fq.gz"]
}]
```

**Reply by asl** [reply]
Yes
